### PR TITLE
Fix sonarcloud vulnerabilites

### LINKED ISSUE #3330: 10 vulnerabilities in Sonarcloud
Here is the [list](https://sonarcloud.io/project/issues?resolved=false&types=VULNERABILITY&id=Clinical-Genomics_scout) on Sonarcloud

Type: **Change this code to not log user-controlled data.**

- [ ]  scout/server/app.py, line 302: `current_app.logger.info("using session language: %s", session_language)`
- [ ] scout/server/blueprints/alignviewers/controllers.py, line 31: `LOG.warning(f"{resource} not in {session.get('igv_tracks', [])}")`
- [ ] scout/server/blueprints/variant/views.py, line 172: `LOG.warning("Attempt to update manual rank with invalid value {}".format(manual_rank))`
- [ ] scout/server/blueprints/variant/views.py, line: 190: `LOG.warning("Attempt to update cancer tier with invalid value {}".format(cancer_tier))`
- [ ]  scout/server/blueprints/variants/views.py, line 214: `return dict blablabla`
- [ ]  scout/server/blueprints/variants/views.py, line 289: `return dict blablabla`
- [ ] scout/server/blueprints/variants/views.py, line 400: `return dict blablabla`
- [ ] scout/server/blueprints/variants/views.py, line 473: `return dict blablabla`

Type: **Change this code to not construct the path from user-controlled data**
- [ ] scout/server/blueprints/cases/views.py, line 263: 
```
temp_excel_dir = os.path.join(cases_bp.static_folder, "_".join([case_name, "mt_reports"]))
os.makedirs(temp_excel_dir, exist_ok=True)
```
Type: **Change this code to use a stronger protocol**
- [ ] scout/server/extensions/ldap_extension.py, line 31: `app.config.setdefault("LDAP_TLS_VERSION", ssl.PROTOCOL_TLSv1)`

### PR DESCRIPTION
**Pull request by northwestwitch**
Fix the 10 code vulnerabilities flagged by SonarCloud (Fix #3330)

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
</details>

**How to test**:
1. 

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

File: scout/server/blueprints/cases/views.py [hunk 1]
```diff
@@ -259,7 +259,9 @@ def mt_report(institute_id, case_name):
     institute_obj, case_obj = institute_and_case(store, institute_id, case_name)
 
     # create a temp folder to write excel files into
-    temp_excel_dir = os.path.join(cases_bp.static_folder, "_".join([case_name, "mt_reports"]))
+    temp_excel_dir = os.path.join(
+        cases_bp.static_folder, "_".join([case_obj["display_name"], "mt_reports"])
```
Comment (dnil) [inline scout/server/blueprints/cases/views.py:263]: 👍

**Review by dnil — APPROVED** [review]

File: scout/server/blueprints/alignviewers/controllers.py [hunk 2]
```diff
@@ -28,7 +28,7 @@ def check_session_tracks(resource):
         LOG.warning("Unauthenticated user requesting resource via remote_static")
         return False
     if resource not in session.get("igv_tracks", []):
-        LOG.warning(f"{resource} not in {session.get('igv_tracks', [])}")
+        LOG.warning(f"Requested resource to be displayed in IGV not in session's IGV tracks")
```
[the remark this answers is not in the document]
Comment (northwestwitch) [reply scout/server/blueprints/alignviewers/controllers.py:31]: It's not this biggie I think..

File: scout/server/extensions/ldap_extension.py [hunk 3]
```diff
@@ -28,7 +28,7 @@ def init_app(self, app):
 
         app.config.setdefault("LDAP_USE_SSL", False)
         app.config.setdefault("LDAP_USE_TLS", True)
-        app.config.setdefault("LDAP_TLS_VERSION", ssl.PROTOCOL_TLSv1)
+        app.config.setdefault("LDAP_TLS_VERSION", ssl.PROTOCOL_TLSv1_2)
```
[the remark this answers is not in the document]
Comment (northwestwitch) [reply scout/server/extensions/ldap_extension.py:31]: Me neither, but the protocol exists, I've checked on the lib manual

**Comment by northwestwitch** [comment]
Ok, looks like my PR [missed 4 vulnerabilities](https://sonarcloud.io/project/issues?resolved=false&types=VULNERABILITY&id=Clinical-Genomics_scout), all of this type:

image

I should have Markdown escaped also variant_type, together with page!

**Comment by dnil** [comment]
Yeah, makes more sense actually - they would be more somewhat more sensitive. Again, it would have to be a devious string indeed to retain any devilry after the `int()`. A really evil Gödel number perhaps?

**Comment by northwestwitch** [comment]
> A really evil Gödel number perhaps?

Satan has its ways!

**Comment by dnil** [comment]
Did you test it btw? Im having trouble getting it on hasta stage - some conflict with jinja2 and markupsafe..

**Comment by northwestwitch** [comment]
> Did you test it btw? Im having trouble getting it on hasta stage - some conflict with jinja2 and markupsafe..

Locally, sorry! what does it say?

**Comment by dnil** [comment]
```
Traceback (most recent call last):
  File "/home/proj/bin/conda/envs/S_scout/bin/scout", line 5, in <module>
    from scout.commands import cli
  File "/home/proj/bin/conda/envs/S_scout/lib/python3.7/site-packages/scout/commands/__init__.py", line 1, in <module>
    from .base import cli
  File "/home/proj/bin/conda/envs/S_scout/lib/python3.7/site-packages/scout/commands/base.py", line 25, in <module>
    from scout.server.app import create_app
  File "/home/proj/bin/conda/envs/S_scout/lib/python3.7/site-packages/scout/server/app.py", line 15, in <module>
    from .blueprints import (
  File "/home/proj/bin/conda/envs/S_scout/lib/python3.7/site-packages/scout/server/blueprints/cases/__init__.py", line 2, in <module>
    from .views import cases_bp
  File "/home/proj/bin/conda/envs/S_scout/lib/python3.7/site-packages/scout/server/blueprints/cases/views.py", line 36, in <module>
    from . import controllers
  File "/home/proj/bin/conda/envs/S_scout/lib/python3.7/site-packages/scout/server/blueprints/cases/controllers.py", line 43, in <module>
    from scout.server.blueprints.variants.controllers import get_manual_assessments
  File "/home/proj/bin/conda/envs/S_scout/lib/python3.7/site-packages/scout/server/blueprints/variants/__init__.py", line 2, in <module>
    from .views import variants_bp
  File "/home/proj/bin/conda/envs/S_scout/lib/python3.7/site-packages/scout/server/blueprints/variants/views.py", line 23, in <module>
    from . import controllers
  File "/home/proj/bin/conda/envs/S_scout/lib/python3.7/site-packages/scout/server/blueprints/variants/controllers.py", line 44, in <module>
    from .forms import (  # noqa: F401; noqa: F401
  File "/home/proj/bin/conda/envs/S_scout/lib/python3.7/site-packages/scout/server/blueprints/variants/forms.py", line 5, in <module>
    from flask_wtf import FlaskForm
  File "/home/proj/bin/conda/envs/S_scout/lib/python3.7/site-packages/flask_wtf/__init__.py", line 4, in <module>
    from .form import FlaskForm, Form
  File "/home/proj/bin/conda/envs/S_scout/lib/python3.7/site-packages/flask_wtf/form.py", line 4, in <module>
    from jinja2 import Markup
ImportError: cannot import name 'Markup' from 'jinja2' (/home/proj/bin/conda/envs/S_scout/lib/python3.7/site-packages/jinja2/__init__.py)
```
I'll try some of the usual culprits..

**Comment by dnil** [comment]
Ok, updating `jinja2`, `flask-wtf` and so on got to the stage where it was only the known issue with `flask-login` (`safe_str_cmp`). Updating it to 0.6.1 got the stage env accepting the branch.

**Comment by dnil** [comment]
It may be a bumpy road on the next hasta upgrade, so we better allocate a bit of time for it. 😅

**Comment by northwestwitch** [comment]
> It may be a bumpy road on the next hasta upgrade, so we better allocate a bit of time for it. 😅

Yes, definitely

**Comment by dnil** [comment]
For posteriority, here is a good pip freeze from local:
```
pip freeze
anytree==2.8.0
attrs==21.4.0
Authlib==1.0.1
Babel==2.10.1
black==22.3.0
blinker==1.4
cairocffi==1.3.0
CairoSVG==2.5.2
certifi==2021.10.8
cffi==1.15.0
charset-normalizer==2.0.12
click==8.1.3
coloredlogs==14.0
configobj==5.0.6
coverage==6.3.2
cryptography==3.3.2
cssselect2==0.6.0
cyvcf2==0.30.15
defusedxml==0.7.1
dominate==2.6.0
Flask==2.1.2
Flask-Babel==2.0.0
Flask-Bootstrap==3.3.7.1
Flask-Cors==3.0.10
Flask-LDAPConn==0.10.1
Flask-Login==0.6.1
Flask-Mail==0.9.1
Flask-Markdown==0.3
Flask-WTF==1.0.1
ghp-import==2.1.0
humanfriendly==10.0
idna==3.3
importlib-metadata==4.11.3
iniconfig==1.1.1
intervaltree==3.0.2
invoke==1.7.0
itsdangerous==2.1.2
Jinja2==3.1.2
ldap3==2.9.1
livereload==2.6.3
lxml==4.8.0
Markdown==3.3.6
markdown-include==0.6.0
MarkupSafe==2.1.1
mergedeep==1.3.4
mkdocs==1.3.0
mkdocs-material==8.2.13
mkdocs-material-extensions==1.0.3
mongomock==3.20.0
mypy-extensions==0.4.3
numpy==1.22.3
packaging==21.3
path==16.4.0
path.py==12.5.0
pathlib==1.0.1
pathspec==0.9.0
pdfkit==1.0.0
ped-parser==1.6.6
Pillow==9.1.0
platformdirs==2.5.2
pluggy==1.0.0
py==1.11.0
pyasn1==0.4.8
pycparser==2.21
pydantic==1.9.0
Pygments==2.12.0
pymdown-extensions==9.4
pymongo==3.12.3
pyparsing==3.0.8
pytest==7.1.2
pytest-cov==3.0.0
pytest-flask==1.2.0
pytest-mock==3.7.0
pytest-test-groups==1.0.3
python-dateutil==2.8.2
pytz==2022.1
PyYAML==6.0
pyyaml_env_tag==0.1
query-phenomizer==1.2
reportlab==3.6.9
requests==2.27.1
responses==0.20.0
-e git+ssh://git@github.com/Clinical-Genomics/scout.git@ce9a089d1bcb688e554854811a1cec2363199b59#egg=scout_browser
sentinels==1.0.0
six==1.16.0
sortedcontainers==2.4.0
svglib==1.2.1
tabulate==0.8.9
tinycss2==1.1.1
tomli==2.0.1
tornado==6.1
typing_extensions==4.2.0
urllib3==1.26.9
visitor==0.1.3
watchdog==2.1.7
webencodings==0.5.1
Werkzeug==2.1.2
WTForms==2.3.3
XlsxWriter==3.0.3
zipp==3.8.0
```

**Pull request merged**